pkg/apis/agones/v1/common.go

```diff
@@ -88,41 +81,20 @@ type gsSpec interface {
 
 // validateGSSpec Check GameServerSpec of a CRD
 // Used by Fleet and GameServerSet
-func validateGSSpec(apiHooks APIHooks, gs gsSpec) []metav1.StatusCause {
+func validateGSSpec(apiHooks APIHooks, gs gsSpec, fldPath *field.Path) field.ErrorList {
 	gsSpec := gs.GetGameServerSpec()
 	gsSpec.ApplyDefaults()
-	causes, _ := gsSpec.Validate(apiHooks, "")
-
-	return causes
+	allErrs := gsSpec.Validate(apiHooks, "", fldPath)
+	return allErrs
 }
 
 // validateObjectMeta Check ObjectMeta specification
 // Used by Fleet, GameServerSet and GameServer
-func validateObjectMeta(objMeta *metav1.ObjectMeta) []metav1.StatusCause {
-	var causes []metav1.StatusCause
-
-	errs := metav1validation.ValidateLabels(objMeta.Labels, field.NewPath("labels"))
-	if len(errs) != 0 {
-		for _, v := range errs {
-			causes = append(causes, metav1.StatusCause{
-				Type:    metav1.CauseTypeFieldValueInvalid,
-				Field:   "labels",
-				Message: v.Error(),
-			})
-		}
-	}
-	errs = apivalidation.ValidateAnnotations(objMeta.Annotations,
-		field.NewPath("annotations"))
-	if len(errs) != 0 {
-		for _, v := range errs {
-			causes = append(causes, metav1.StatusCause{
-				Type:    metav1.CauseTypeFieldValueInvalid,
-				Field:   "annotations",
-				Message: v.Error(),
-			})
-		}
-	}
-	return causes
+func validateObjectMeta(objMeta *metav1.ObjectMeta, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
```

Comment:
nit: here you declare var and then append to it, whereas on line 114 below you start with `allErrs := `. I actually prefer the line 114 version since it's one fewer line, but regardless I would rather they be consistent. :)

Reply:
This is my omission, it's done